cmd/api/src/database/agi.go

```diff
@@ -87,46 +86,58 @@ func (s *BloodhoundDB) DeleteAssetGroup(ctx context.Context, assetGroup model.As
 }
 
 func (s *BloodhoundDB) GetAssetGroup(ctx context.Context, id int32) (model.AssetGroup, error) {
-	var (
-		assetGroup model.AssetGroup
-		result     = s.preload(model.AssetGroupAssociations()).WithContext(ctx).First(&assetGroup, id)
-	)
-	return assetGroup, CheckError(result)
+    var assetGroup model.AssetGroup
+    if result := s.preload(model.AssetGroupAssociations()).WithContext(ctx).First(&assetGroup, id); errors.Is(result.Error, gorm.ErrRecordNotFound) {
+        return assetGroup, ErrNotFound
+    } else if result.Error != nil {
+        return assetGroup, result.Error
+    }
+
+    latestCollection, collectionErr := s.GetLatestAssetGroupCollection(ctx, id)
+    if collectionErr != nil {
+        if errors.Is(collectionErr, ErrNotFound) {
+            assetGroup.MemberCount = 0
+            return assetGroup, nil
+        }
+        return assetGroup, fmt.Errorf("error getting latest collection for asset group %s: %w", assetGroup.Name, collectionErr)
+    }
+    assetGroup.MemberCount = len(latestCollection.Entries)
+
+    return assetGroup, nil
```

Comment:
This could be rewritten to chain with the previous block:

```go
    else if latestCollection, collectionErr := s.GetLatestAssetGroupCollection(ctx, id); errors.Is(collectionErr, ErrNotFound) {
        assetGroup.MemberCount = 0
        return assetGroup, nil
    } else if collectionErr != nil {
        return assetGroup, fmt.Errorf("error getting latest collection for asset group %s: %w", assetGroup.Name, collectionErr)
    } else {
        assetGroup.MemberCount = len(latestCollection.Entries)
        return assetGroup, nil
    }
```